lib/src/Recorder.cpp

```diff
@@ -221,6 +220,8 @@ void Recorder::recordingFileReopen()
 
 void Recorder::startRecording()
 {
+    MUTEX();
```

Comment:
Could cause a deadlock. Later while calling `recordLine(...)`, it calls `MUTEX()` again. 

From https://en.cppreference.com/w/cpp/thread/mutex, std::mutex offers exclusive, non-recursive ownership semantics.

Reply:
Good catch, I've enclosed the lock operation and all the direct ofstream operations in a scoped block to avoid this.